### ISSUE TITLE
Be able to specify suffix for temporary file

### ISSUE THREAD

**Issue by elemoine** [geopython/pywps#242]
The command I need to wrap into a WPS Process requires that the file names it is passed as input arguments have a specific extension (`.pnm`).

Currently there's no way to specify suffixes for temporary files created by `IOHandler.get_file`.  This issues suggests to extend `IOHandler` and its users to make that possible.

For example, this is what I'd like to be able to do:

```py
inputs = [
    ComplexInput('i', 'image',
                 supported_formats=[Format('image/image/x-portable-anymap')],
                 tempfile_suffix='.pnm')
]
```

Does that sound reasonable? Do you have other suggestions?

Thanks.

**Comment by jachym** [comment]
Hi, that is reasonable reature request. Are you able to dig into this by yourself? Would https://github.com/geopython/pywps/blob/develop/pywps/inout/basic.py#L145 be the line, you are looking for? Of course, we would have to propagate the attribute all the way down to the constructor https://github.com/geopython/pywps/blob/develop/pywps/inout/basic.py#L80 but that should be possible - what do you think?

**Comment by elemoine** [comment]
Yep. Thank you for the note. I'll create a PR for this.

**Comment by cehbrecht** [comment]
From my perspective i have processes which do not only rely on the file extension but also on the filename itself. My wish list would be:

* when a ``ComplexInput`` is provided as url the filename part with extension should be kept (``http://somewhere/path/to/myfile.png  -> myfile.png``). If there are name collisions then the name will be extended ... like with tempfile keeping the filename (without extension) as prefix and the file extension (``myfile-aytgh.png``).
* when the ``ComplexInput`` has no url/filename (base64 included in request) then a tempfile needs to be generated ... as @elemoine said, if would be nice to have an appropriate file extension (``fgbui.png``).

We already have the mime-type formats. Could we associate a default extension to these mime-types which can be used for creating the temp filenames ... and/or add the file extension to the ``Format`` as optional parameter (``Format("image/png", extension=".png")``)?

**Comment by elemoine** [comment]
@cehbrecht thanks for your comments. Did you see https://github.com/geopython/pywps/pull/243? In this patch the extension is specified on the `ComplexInput` itself rather than on the `Format` as you are suggesting. I like your idea of having sensible defaults and storing the extension in the `Format` but I'll have to check that the `Format` object is available when creating the temporary file.

**Comment by cehbrecht** [comment]
@elemoine i saw the PR ... i was a bit late bringing in my comments ;) 

I haven't checked if it will work with the ``Format`` object ... and i would leave the decision to the pywps people what they prefer ...

**Comment by jachym** [comment]
@cehbrecht there is actually data for mimetype->extension asignment in `pywps/inout/formats/__init__.py` could be re-used

**Comment by elemoine** [comment]
> @cehbrecht there is actually data for mimetype->extension asignment in pywps/inout/formats/__init__.py could be re-used

Thanks. I'll have a look.

**Comment by ldesousa** [comment]
Fixed in PR #258 and PR #259
